### PR TITLE
Describe js-implicit parameters

### PR DESCRIPTION
**Pull request by ripsawridge**
Hi Tobias, this commit is correctly associated with my chromium.org account.

### CODE REVIEW

**Comment by ripsawridge** [comment]
I checked in a fix for the build failure.

File: src/docs/torque.md [hunk 1]
```diff
@@ -503,10 +503,29 @@ macro Bar(implicit context: Context)() {
...
+
+Since overload resolution can cause confusing behavior, we ensure that implicit parameters do not influence overload resolution at all. That is: when comparing candidates of an overload set, we do not consider the available implicit bindings at the call-site. Only after we found a single best overload, we check if implicit bindings for the implicit parameters are available.
 
 Having the implicit parameters left of the explicit parameters is different from Scala, but maps better to the existing convention in CSA to have the `context` parameter first.
 
+#### js-implicit
+
+If you want JavaScript linking for a builtin defined in Torque, you should use keyword `js-implicit` instead of `implicit`. The arguments are limited to these four components of the calling convention:
+
+- context: Context
```
Comment (tebbi) [inline src/docs/torque.md]: This js-implicit context has type NativeContext now. Perhaps we should explain that this is due to builtins embedding a NativeContext as their closure context.

File: src/docs/torque.md [hunk 2]
```diff
@@ -503,10 +503,29 @@ macro Bar(implicit context: Context)() {
...
+- receiver: Object ("this" in JavaScript)
```
Comment (tebbi) [inline src/docs/torque.md]: receiver has type JSAny

File: src/docs/torque.md [hunk 3]
```diff
@@ -503,10 +503,29 @@ macro Bar(implicit context: Context)() {
...
+- target: JSFunction (arguments.callee in JavaScript)
+- newTarget: Object (new.target in JavaScript)
```
Comment (tebbi) [inline src/docs/torque.md]: newTarget is JSAny too.

File: src/docs/torque.md [hunk 4]
```diff
@@ -503,10 +503,29 @@ macro Bar(implicit context: Context)() {
...
+
+They don't all have to be declared, only the ones you want to use. For an example, here is our code for `Array.prototype.shift`:
+
+```torque
+  // https://tc39.github.io/ecma262/#sec-array.prototype.shift
+  transitioning javascript builtin ArrayPrototypeShift(
+      js-implicit context: Context, receiver: Object)(...arguments): Object {
```
Comment (tebbi) [inline src/docs/torque.md]: transitioning javascript builtin ArrayPrototypeShift(
      js-implicit context: NativeContext, receiver: JSAny)(...arguments): JSAny {

File: src/docs/torque.md [hunk 5]
```diff
@@ -503,10 +503,29 @@ macro Bar(implicit context: Context)() {
...
-For overload resolution not to cause fragile and confusing behavior, we want that the implicit parameters do not influence overload resolution at all. That is: when comparing candidates of an overload set, we do not consider the available implicit bindings at the call-site. Only after we found a single best overload, we check if implicit bindings for the implicit parameters are available.
+
+Since overload resolution can cause confusing behavior, we ensure that implicit parameters do not influence overload resolution at all. That is: when comparing candidates of an overload set, we do not consider the available implicit bindings at the call-site. Only after we found a single best overload, we check if implicit bindings for the implicit parameters are available.
 
 Having the implicit parameters left of the explicit parameters is different from Scala, but maps better to the existing convention in CSA to have the `context` parameter first.
 
+#### js-implicit
+
+If you want JavaScript linking for a builtin defined in Torque, you should use keyword `js-implicit` instead of `implicit`. The arguments are limited to these four components of the calling convention:
```
Comment (tebbi) [inline src/docs/torque.md]: This sounds a bit like js-implicit causes JavaScript linkage. Perhaps we should say something like:
"For builtins with JavaScript linkage defined in Torque, you ..."

File: src/docs/torque.md (hunk 1 above)
Comment (ripsawridge) [reply src/docs/torque.md]: Right on, done.

File: src/docs/torque.md (hunk 5 above)
Comment (ripsawridge) [reply src/docs/torque.md]: Good point. done.

**Comment by ripsawridge** [comment]
Thanks Tobias, I addressed your comments and will merge.

**Pull request merged**